packages/nimble-components/build/generate-workers/source/matrix-renderer.ts

```diff
@@ -8,16 +9,128 @@ import { expose } from 'comlink';
  * This setup is used in the wafer-map component to perform heavy computational duties
  */
 export class MatrixRenderer {
-    public dieMatrix: Uint8Array = Uint8Array.from([]);
+    public colIndexes: Uint32Array = Uint32Array.from([]);
+    public rowIndexes: Uint32Array = Uint32Array.from([]);
+    public values = new Float64Array([]);
+    public scaledColIndex = new Float64Array([]);
+    public scaledRowIndex = new Float64Array([]);
+    public canvas!: OffscreenCanvas;
+    public context!: OffscreenCanvasRenderingContext2D;
+    private scaleX!: number;
+    private scaleY!: number;
+    private baseX!: number;
+    private baseY!: number;
+    private dieDimensions!: Dimensions;
+    private transform!: Transform;
+    private topLeftCanvasCorner!: { x: number; y: number; };
```

Comment:
Yeah, it's not proper to declare everything as a non-nullable value if they are not at least initialized in the constructor (or in some other specific use-cases like elements that are defined in a FAST template). To me, the alternative is to provide expected defaults for some of the values (like `scaleX` and `scaleY` which I presume can be 1), and for others where there is no obvious default (like `topLeftCanvasCorner`) then the two choice are:
- Provide the values as constructor parameters (and note that there could even be a data structure that makes sense to create that captures all of the necessary initial values in one object).
    - This option doesn't seem applicable with the way the `MatrixRenderer` is initialized in practice.
- Leave the values as nullable, which then requires appropriate checks in the code against those fields.